src/libasr/pass/intrinsic_functions.h

```diff
@@ -1384,6 +1388,142 @@ namespace Ble {
+    }
+
+} // namespace Llt
+
+namespace Lge {
+
+    static ASR::expr_t *eval_Lge(Allocator &al, const Location &loc,
+        ASR::ttype_t* t1, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        char* string_A = ASR::down_cast<ASR::StringConstant_t>(args[0])->m_s;
+        char* string_B = ASR::down_cast<ASR::StringConstant_t>(args[1])->m_s;
+        bool result = false;
+        if (strcmp(string_A, string_B) >= 0) {
+            result = true;
+        }
+        return make_ConstantWithType(make_LogicalConstant_t, result, t1, loc);
+    }
+
+    static inline ASR::expr_t* instantiate_Lge(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_lge_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", ASRUtils::TYPE(ASR::make_Character_t(al, loc, 1, -1, nullptr)));
+        fill_func_arg("y", ASRUtils::TYPE(ASR::make_Character_t(al, loc, 1, -1, nullptr)));
+        auto result = declare(fn_name, return_type, ReturnVar);
+        body.push_back(al, b.If(sGtE(args[0], args[1]), {
+            b.Assignment(result, b.bool32(1))
+        }, {
+            b.Assignment(result, b.bool32(0))
+        }));
+
+        ASR::symbol_t *f_sym = make_ASR_Function_t(fn_name, fn_symtab, dep, args,
+            body, result, ASR::abiType::Source, ASR::deftypeType::Implementation, nullptr);
+        scope->add_symbol(fn_name, f_sym);
+        return b.Call(f_sym, new_args, logical, nullptr);
+    }
+
+} // namespace Lge
+
+namespace Lle {
+
+    static ASR::expr_t *eval_Lle(Allocator &al, const Location &loc,
+        ASR::ttype_t* t1, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        char* string_A = ASR::down_cast<ASR::StringConstant_t>(args[0])->m_s;
+        char* string_B = ASR::down_cast<ASR::StringConstant_t>(args[1])->m_s;
+        bool result = false;
+        if (strcmp(string_A, string_B) <= 0) {
+            result = true;
+        }
+        return make_ConstantWithType(make_LogicalConstant_t, result, t1, loc);
+    }
+
+    static inline ASR::expr_t* instantiate_Lle(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_lle_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", ASRUtils::TYPE(ASR::make_Character_t(al, loc, 1, -1, nullptr)));
+        fill_func_arg("y", ASRUtils::TYPE(ASR::make_Character_t(al, loc, 1, -1, nullptr)));
+        auto result = declare(fn_name, return_type, ReturnVar);
+        body.push_back(al, b.If(sLtE(args[0], args[1]), {
+            b.Assignment(result, b.bool32(1))
+        }, {
+            b.Assignment(result, b.bool32(0))
+        }));
```

Comment:
Can we not do something like?

```suggestion
	body.push_back(al, b.Assignment(result, sLtE(args[0], args[1])));
```

Reply:
Try this locally, if it works use it at everyplace. I am converting it to draft.